src/Database/Schema/Column.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace TCG\Voyager\Database\Schema;
+
+use Doctrine\DBAL\Schema\Column as DoctrineColumn;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Schema\Comparator;
+
+class Column
```

Comment:
The reason I am passing this to the constructor is that Doctrine already has methods to retrieve and setup columns from the database, and extending means that I will have to create special methods to populate the columns from the database, and that will add more complexity.
Anyways, I have actually stopped working on this. I am currently working on a much simpler approach that does what need instead of trying to do many things and end up overcomplicating the whole thing. I will close this PR and publish the new one tomorrow.